src/main/java/sparksoniq/jsoniq/runtime/iterator/operational/TreatIterator.java

```diff
@@ -23,6 +23,7 @@
     private static final long serialVersionUID = 1L;
     private RuntimeIterator _iterator;
     private Item _nextResult;
+    private Item _currentResult;
```

Comment:
A cosmetic (and, to some extent, performance-related) comment: since the only test involving this variable is whether it is null, a boolean may suffice.

Reply:
Using the _\_currentResult_ item, we can also avoid creating a new short-lived _Item result_ object in every iteration of the loop, and use _\_currentResult_ to store the value of _nextResult_ before the next() call. I agree that we can also go back to using a new inside-loop variable as before and using a boolean to check whether we are in the first iteration of the loop. I have no strong preference for either of the two options here. If you feel the latter is better, I will do that.

Reply:
Your reasoning makes a lot of sense. I like to see this as an optimization.